src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs

```diff
@@ -322,6 +322,70 @@ public IEnumerable<TabInfo> GetPageList(int parentId = -1, string searchKey = ""
             return pages;
         }
 
+        private IEnumerable<TabInfo> GetPageListHelper(string searchKey = "", bool includeHidden = true, bool includeDeleted = false)

```

Comment:
This method is very similar to `public IEnumerable<TabInfo> GetPageList`. I see two differences: 

1. no `(t.ParentId == parentId)` check
1. added sorting

The first is fine but what about the second? Why add sorting where the other does not do ordering?

Would you try to extract a common denominator?

Reply:
Also, not about your code per se, but the multi-line `where` statement is mighty awful. 

Please refactor this. I'd start with extracting parts of that expression into [`let` statements](https://docs.microsoft.com/dotnet/csharp/language-reference/keywords/let-clause)